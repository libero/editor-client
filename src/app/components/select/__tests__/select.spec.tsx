import React from 'react';
import { create } from 'react-test-renderer';

import { Select } from '..';

describe('Select component', () => {
  it('should render component', () => {
    const wrapper = create(
      <Select
        className={'test-class-name'}
        name="articleType"
        placeholder="Please select"
        fullWidth
        blankValue={undefined}
        label="Competing interest"
        value={false}
        onChange={jest.fn()}
        options={[
          { label: 'No competing interest', value: false },
          { label: 'Has competing interest', value: true }
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
