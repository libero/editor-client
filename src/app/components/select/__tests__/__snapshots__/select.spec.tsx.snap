// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render component 1`] = `
<div
  className="MuiFormControl-root test-class-name MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
    data-shrink={true}
    id="select-id-0"
  >
    Competing interest
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="select-id-0 mui-component-select-articleType"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      id="mui-component-select-articleType"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      No competing interest
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="articleType"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="select-id-1"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
      >
        <span>
          Competing interest
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
